packages/typespec-test/test/eventgrid_modular/generated/typespec-ts/src/api/operations.ts

```diff
@@ -345,14 +346,17 @@ export function _publishCloudEventsSend(
       allowReserved: options?.requestOptions?.skipUrlEncoding,
     },
   );
-  return context.path(path).post({
-    ...operationOptionsToRequestParameters(options),
-    contentType: "application/cloudevents-batch+json; charset=utf-8",
-    headers: { accept: "application/json", ...options.requestOptions?.headers },
-    body: events.map((p: any) => {
-      return cloudEventSerializer(p);
-    }),
-  });
+  return context
+    .path(path)
+    .post({
+      ...operationOptionsToRequestParameters(options),
+      contentType: "application/cloudevents-batch+json; charset=utf-8",
+      headers: {
+        accept: "application/json",
+        ...options.requestOptions?.headers,
+      },
+      body: cloudEventArraySerializer(events),
```

Comment:
@qiaozha could you help advise if this is introduce by tcgc with the issue fix: https://github.com/Azure/typespec-azure/issues/2595?